### PR TITLE
Fixes stock price spikes

### PR DESCRIPTION
**Pull request by Iamgoofball**
i'll admit this meme is bandaid as FUCK but it'll also help us diagnose the problem

### CODE REVIEW

**Comment by ma44** [comment]
HE DID IT BOIS
https://github.com/tgstation/tgstation/issues/16598

**Comment by Iamgoofball** [comment]
#20018

File: code/modules/stock_market/stocks.dm [hunk 1]
```diff
@@ -139,8 +139,16 @@
...
-
-	current_value += (speculation / rand(25000, 50000) + performance / rand(100, 800)) * current_value
+	var/fucking_stock_spikes = current_value + 500
+	var/piece_of_shit_fuck = current_value - 500
+	var/i_hate_this_code = (speculation / rand(25000, 50000) + performance / rand(100, 800)) * current_value
+	if(i_hate_this_code > fucking_stock_spikes || i_hate_this_code < piece_of_shit_fuck)
+		message_admins("STOCK ALERT: [name] attempted to increase/decrease its value by more than 500 credits, this has been stopped. Please alert iamgoofball on #coderbus at irc.rizon.net with the following info:")
+		message_admins("STOCK ALERT: Speculation was: [speculation]")
+		message_admins("STOCK ALERT: Performance was: [performance]")
+		message_admins("STOCK ALERT: If you can, take a screenshot of this data.")
```
Comment (Jordie0608) [inline code/modules/stock_market/stocks.dm]: This is going to be obtuse to admins and difficult for you to ever analyze.
It'd be all around better to do what was discussed before: send these to an investigate log that's saved at round-end so the info is both self-contained and easy to interpret.
You may also want to include a timestamp and the stock transaction log made by players.

Comment (Iamgoofball) [inline code/modules/stock_market/stocks.dm]: i took a look at shit and the only thing that can cause this is the automatic market fluctuation, which means player purchases don't matter.

i'll look into a file

**Comment by ExcessiveUseOfCobblestone** [comment]
This [the stocks getting so big they go into e+XX] has only happened to me when I flooded Cargo with points [via $lemons], then proceeded to flood a certain stock buy buying thousands of said stock at once.

**Comment by Iamgoofball** [comment]
NOBODY FUCKING TOLD ME THIS

**Comment by optimumtact** [comment]
So apparently this doesn't happpen?

**Comment by bgobandit** [comment]
It definitely happen**ed** at one point, though it was hard to diagnose

**Comment by Nyvrem** [comment]
I had plenty of rounds where it happened randomly, although it seemed likelier to occur with stocks in companies that were about to go bankrupt.

**Comment by PKPenguin321** [comment]
wait so if it happened when they were about to go bankrupt could that mean that it went negative and rolled over or something

**Comment by Nyvrem** [comment]
I don't believe it was the only case in which it happened, although it sure as hell seemed more likely to occur in such a situation.

**Pull request merged**